### ISSUE TITLE
Confidence intervals using bootstrap vs estimated sigma

### ISSUE THREAD

**Issue by nicolasch96** [gfxdisp/asap#10]
Hello,

We are still using your paper for some testing, and we were trying to compute some confidence intervals through bootstrapping. The work is done with @SamuelSS09. 

As mentioned in your article, the estimated $sigma$ is the standard error of the sample of observers, i.e. the std deviation of the estimated score for a certain image.

From bootstrapping for a random image, we get these results for a different number of observers (10 to 25):
![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/48471976/144620607-91938f8c-d452-41b6-a6b2-7e403bcd7e93.png)

To compute the confidence interval, we proceed to random sampling with replacement and then take the std of the predicted score over 1000 repetitions and compute the confidence interval at 95% (z=1.96). we also verify using the percentile method and we get the same results. Now, if we use the algorithm std, we usually get 2x1.96xstd which gives us different values. For example, for a point where the confidence interval is equal to 1.9, the std is around 0.3 ==> confidence interval from that is 0.3x2x1.96=1.176, which is a little different.

As we see, the sigma is not correlated well with the bootstrap confidence interval, is this normal? I would love guidance from your part.

Thank you very much in advance!

Best,

Nicolas

**Comment by mikhailiuk** [comment]
Hi Nicolas,

As always thanks a lot for using the method! And raising very sensible and practical issues :)

I actually conducted a similar experiment to verify if the std of ASAP is similar to the one computed by bootstrap. 

I will need some time to get hold of the procedure and results as it has been performed almost two years ago..

I will get back to you towards the end of the next week.

Thanks,

Aliaksei

**Comment by nicolasch96** [comment]
Thank you very much for your fast response! I really appreciate this work!

I know it's all estimations and practice can differ immensely, but I just want to make sure to use the sigma of the algorithm in the correct way!

Good luck with your future projects!

Best,

Nicolas

**Comment by mikhailiuk** [comment]
Hey!

Copying the points we discussed on LinkedIn here:

The experiment on CIs that I have performed was in simulation (I have not checked CIs for very sparse matrices). 

Trueskill doesn’t have the notion of observers, so you’d need to bootstrap individual outcomes to get CIs comparable to the ones estimated by the algorithm.

For example let’s imagine that we have two observers each making three judgements for three objects.  

[0,1,0; 
1,0,0; 
0,1,0]  

[0,1,1; 
0,0,0; 
1,0,0] 

If we sample observers with bootstrapping we would be sampling two rows with replacement, i.e. unfolded matrices:  

[0,1,0,1,0,0,0,1,0] 
[0,1,1,0,0,0,1,0,0]  

If we sample single outcomes, we would separate the rows into individual trial outcomes:  

[0,1,0,0,0,0,0,0,0] 
[0,0,0,1,0,0,0,0,0] 
[0,0,0,0,0,0,0,1,0] 
[0,1,0,0,0,0,0,0,0] 
[0,0,1,0,0,0,0,0,0] 
[0,0,0,0,0,0,1,0,0] 

In the second case bootstrapping with resampling should produce comparable CIs to the once produced by ASAP (with larger number of trials it converges).  

For sparse matrices this wouldn’t necessarily hold, as CI wouldn’t be very accurate either.
